homeassistant/components/idasen_desk/cover.py

```diff
@@ -54,15 +48,14 @@ class IdasenDeskCover(CoordinatorEntity, CoverEntity):
 
     def __init__(
         self,
-        desk: Desk,
         address: str,
         device_info: DeviceInfo,
-        coordinator: DataUpdateCoordinator,
+        coordinator: IdasenDeskCoordinator,
     ) -> None:
         """Initialize an Idasen Desk cover."""
         super().__init__(coordinator)
-        self._desk = desk
-        self._attr_name = device_info[ATTR_NAME]
+        self._desk: Desk = coordinator.desk
+        self._attr_name = device_info.get(ATTR_NAME)
```

Comment:
Is the name now not always going to be there?

Reply:
It is still always there. I am now using `pyright` and it complains with:
```
Pyright: Could not access item in TypedDict
     "name" is not a required key in "DeviceInfo", so access may result in runtime exception [reportTypedDictNotRequiredAccess]
```

Should I revert it back?

(I can't use pylance since I am not using vscode)

Reply:
vscode doesn't complain about it.

We know its there because you set it in `async_setup_entry` so it would be a bug if it wasn't.  Please revert it back.

Reply:
Looks good to merge once this outstanding review comment is addresesed

Reply:
Oops! I reverted it in https://github.com/home-assistant/core/pull/102135/commits/5c507e02b51955bdccbd63563ce5c0a00fd952db#diff-1b01d895b3a26b42bfc8a33f512b33e8fa163c2b19c8db761bdde72385c4ee03R128 but forgot about this one.

Reply:
Done now